storage/store.go

```diff
@@ -1080,6 +1082,11 @@ func (s *Store) ExecuteCmd(ctx context.Context, call client.Call) error {
 		}
 	}
 
+	// Check the txn cache for push txn request
+	if ok := s.txnCache.processPushedTxn(args, reply); ok {
```

Comment:
I'm not sure what the original intent of #627 was, but @tschottdorf and I were discussing this change and we think it might be better to have a cache on the sender side (in Store.resolveWriteIntentError) instead of the receiver. A sender-side cache would have a lower hit rate but a higher impact when it matched; I'm not sure which is more important. We need a better idea of the justification for this change - what is the scenario where we expect this optimization to be worthwhile? @spencerkimball